### PR TITLE
Add missing assert.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: tests/TestCase/Validation/ValidatorTest.php [hunk 1]
```diff
@@ -630,14 +630,14 @@ public function testAllowEmptyFor()
         $results = $validator->errors(['name' => null]);
         $this->assertSame([], $results);
 
-        $this->assertSame([], $results);
         $results = $validator->errors(['name' => '']);
-
         $this->assertSame([], $results);
-        $results = $validator->errors(['title' => 0]);
 
+        $results = $validator->errors(['title' => 0]);
```
Comment (othercorey) [inline tests/TestCase/Validation/ValidatorTest.php]: Did you mean 'name' here?

File: tests/TestCase/Validation/ValidatorTest.php [hunk 2]
```diff
@@ -630,14 +630,14 @@ public function testAllowEmptyFor()
...
         $this->assertSame(['title' => ['minLength' => 'Min. length 5 chars']], $results);
+
         $results = $validator->errors(['title' => []]);
+        $this->assertSame([], $results);
```
Comment (ADmad) [inline tests/TestCase/Validation/ValidatorTest.php]: @markstory I think this test shows that the changes I have done in #13904 are actually necessary. Even though `->allowEmptyFor('name', Validator::EMPTY_STRING)` is used for the field it treats empty array as empty and skips validation.

After my changes it runs the validations as expected and generates error https://github.com/cakephp/cakephp/pull/13904/files#diff-2f1913506d5e9a181af4bb924b1a3018R640

File: tests/TestCase/Validation/ValidatorTest.php (hunk 1 above)
Comment (ADmad) [reply tests/TestCase/Validation/ValidatorTest.php]: Doh, I have messed the field names in both PR. Thanks for point it out.

Comment (othercorey) [reply tests/TestCase/Validation/ValidatorTest.php]: I missed it too and merged it :(

**Pull request merged**